### PR TITLE
Make installing krew even simpler

### PR DESCRIPTION
**Pull request by unixorn**
* Eliminate potential copy-pasta errors by adding an install script.
* Make `bashrc`/`zshrc` snippet only add `krew`'s `bin` directory if it exists.
* Update README.md to include new install option.

### CODE REVIEW

**Comment by unixorn** [comment]
Signed the CLA

**Comment by unixorn** [comment]
Trying to retrigger the CLA check

**Comment by ahmetb** [comment]
Nice. While it's tempting, it's not a whole lot different than what we already have :| This script can be `curl | sh`'ed (which is being discussed at #405).

On the other hand, we now have to update not only the readme, but another file while releasing a new version.

**Comment by unixorn** [comment]
Maybe double tag releases as x.y.z and also stable? then the script could just refer to stable, and you wouldn't have to update the readme every release either.

**Comment by ahmetb** [comment]
Instead of re-tagging using the same tag name, we might make use of some redirection through a domain name we own, like krew.dev and create a curl|sh endpoint, but for now it's not a desired installation method and usually frowned upon in Kubernetes community. But more and more, it feels unavoidable.

**Comment by unixorn** [comment]
Or have a STABLE file in the repo with a version number. Then install can be `git clone git@github.com:kubernetes-sigs/krew.git && krew/install-krew`. It's tiny bit more complex than curl|sh, but at least then `git` takes care of verifying the source.

And agreed, curl|sh is ugly, especially given the potential shenanigans shown in https://www.idontplaydarts.com/2016/04/detecting-curl-pipe-bash-server-side/

**Comment by ahmetb** [comment]
Specifically why not “double tag”:

* it’s not a good practice to have floating tags, as they are meant to be snapshots
* we use file download counts on each tag to track krew usage. deleting a tag and re tagging would lose that.

**Comment by unixorn** [comment]
Yeah, double tagging feels kludgier the more we discuss it. A `STABLE_RELEASE` file seems cleaner, but still not perfect.

Better than a raw curl|sh though.

**Comment by ahmetb** [comment]
STABLE_RELEASE would actually work. We’d still need to update windows instructions manually, though.

**Comment by unixorn** [comment]
Ok, I'll update to support that so you can see if it's less ugly

**Comment by unixorn** [comment]
BTW, do you prefer the commits squashed?

**Comment by ahmetb** [comment]
No preference. In fact we don’t even need code (we could have discussed a proposal issue) but it’s ok.

File: install-krew [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+#
+# Install krew
+#
+# License: Apache 2.0 - see https://github.com/kubernetes-sigs/krew/blob/master/LICENSE
+
+set -o pipefail
+set -o errexit
+
+STABLE_RELEASE="$(cat $(dirname $0)/STABLE_RELEASE)"
```
Comment (corneliusweig) [inline install-krew]: You need to fetch the current `STABLE_RELEASE` file from remote. Users should not have to clone krew or download the source tarball for install.

**Comment by corneliusweig** [comment]
I'm not sure I follow correctly. AFAICT, you propose that the installation will still go via an installer script. Unless the goal is to enable the `curl|sh` pattern, I don't see any advantage over the current shell install snippet. I'd even say the UX is worse, because a new user needs to download the script, make it executable and so on.

If the `curl|sh` thing is the goal, then let's continue the discussion on #405.

**Comment by unixorn** [comment]
Downloading is `git clone`, and the script is marked executable in the repository. It's not meant to be a `curl|sh` enabler, I added it to the repo so `git` will automatically handle confirming that it's the correct commit without the end user having to do any SHA checks by hand.

**Comment by corneliusweig** [comment]
I don't think this will improve the UX for new users. New users should be onboarded by downloading the krew binary and setting up the env variables to use krew. If the installation now goes via cloning the repository this makes things more complicated, don't you think?

**Comment by ahmetb** [comment]
+1 this is sadly not improving the experience, it's the same thing in an alternative form, plus there are no _copy-pasta issues_ reported as far as I know.

Let's continue brainstorming at #405. It seems like anything short of curl|sh is not going to make a substantial change.
/close

**Pull request closed without merging**